google/cloud/storage/bucket_metadata.h

```diff
@@ -15,14 +15,71 @@
 #ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_BUCKET_METADATA_H_
 #define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_BUCKET_METADATA_H_
 
+#include "google/cloud/storage/bucket_access_control.h"
 #include "google/cloud/storage/internal/common_metadata.h"
 #include "google/cloud/storage/internal/nljson.h"
+#include "google/cloud/storage/object_access_control.h"
 #include <map>
+#include <tuple>
+#include <utility>
 
 namespace google {
 namespace cloud {
 namespace storage {
 inline namespace STORAGE_CLIENT_NS {
+/**
+ * The billing configuration for a Bucket.
+ *
+ * @see https://cloud.google.com/storage/docs/requester-pays for general
+ *     information on "Requester Pays" billing.
+ */
+struct BucketBilling {
+  bool requester_pays;
+};
+
+/**
+ * An entry in the CORS list.
+ *
+ * CORS (Cross-Origin Resource Sharing) is a mechanism to enable client-side
+ * cross-origin requests. An entry in the configuration has a maximum age and a
+ * list of allowed origin and methods, as well as a list of returned response
+ * headers.
+ *
+ * @see https://en.wikipedia.org/wiki/Cross-origin_resource_sharing for general
+ *     information on CORS.
+ *
+ * @see https://cloud.google.com/storage/docs/cross-origin for general
+ *     information about CORS in the context of Google Cloud Storage.
+ *
+ * @see https://cloud.google.com/storage/docs/configuring-cors for information
+ *     on how to set and troubleshoot CORS settings.
+ */
+struct CorsEntry {
+  std::int64_t max_age_seconds;
```

Comment:
> I think it would be nice to have the ability to differentiate between `0` and "no value present".

nice or necessary? If we think the application developers would need to know the difference then we should (or rather must) do one of these:

- We could use `-1` to represent "not there": the field is signed, and giving a negative value (should) be invalid.
- We could use `0` to represent "not there": semantically `0` and missing are probably the same thing.
- We could use `optional<>`: we have not done this for many fields (I am about to in other places), because they are really messy.  Simple things get a lot more complicated.

Reply:
It seems the C# library makes this Nullable, so an Optional type here seems like it would give us the equivalent functionality and semantics.

Reply:
Done. Let me know what you think of the changes. I had to add additional functionality to `optional<T>` too.